examples/helloworld/src/index.js

```diff
@@ -1,4 +1,26 @@
 import React from 'react';
+import _ from 'underscore';
 import App from './App';
 
-React.render(<App />, document.getElementById('root'))
+
+var state = {
+    very: {
+        deeply: {
+            nested: {
+                counts: _.range(400).map(function () { return 0; })
+            }
+        }
+    }
+};
+
+function swapper(f) {
```

Comment:
Object.observe is es7, only implemented on chrome atm

On 3:33pm, Thu, Sep 24, 2015 Dustin Getz <notifications@github.com> wrote:

> In examples/helloworld/src/index.js
> <https://github.com/dustingetz/react-cursor/pull/63#discussion_r40362175>:
>
> >  import App from './App';
> >
> > -React.render(<App />, document.getElementById('root'))
> > +
> > +var state = {
> > +    very: {
> > +        deeply: {
> > +            nested: {
> > +                counts: _.range(400).map(function () { return 0; })
> > +            }
> > +        }
> > +    }
> > +};
> > +
> > +function swapper(f) {
>
> I don't think it is a good idea to expose such a hard-to-understand
> function to user.
>
> What if we implemented Clojure's atom
> <https://clojuredocs.org/clojure.core/atom> concept, is this any better?
>
> function Atom (initialValue) { this.value = initialValue; }Atom.prototype.setValue = function (nextValue) {
>     var oldValue = this.value;
>     this.value = nextValue;
>     this.dispatchChangeToWatchers(oldValue, this.value);
> }Atom.prototype.addWatch = function (cb) {}
> var stateAtom = new Atom({});
> atom.addWatch((prevVal, val) => queueRender());
> function queueRender() {
>     var cur = ReactCursor.Cursor.build(atom.value, atom.setValue);
>     React.render(<App cursor={cur} />, document.getElementById('root'));
> }
>
> Or even just use Object.observe on a plain old javascript object, though I
> am concerned about the non-determinism of object.observe polyfills since
> they are implemented with polling.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dustingetz/react-cursor/pull/63/files#r40362175>.
>